### PR TITLE
Rebuild index

### PR DESCRIPTION
**Pull request by kulmann**
CLI command for rebuilding the index. Can be called from accounts cli with `rebuild` command.

### CODE REVIEW

File: accounts/pkg/command/rebuild_index.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	"github.com/micro/cli/v2"
+	"github.com/micro/go-micro/v2/client/grpc"
+	merrors "github.com/micro/go-micro/v2/errors"
+	"github.com/owncloud/ocis/accounts/pkg/config"
+	index "github.com/owncloud/ocis/accounts/pkg/proto/v0"
+)
+
+// DeleteIndex rebuilds the entire configured index.
+func DeleteIndex(cdf *config.Config) *cli.Command {
```
Comment (butonic) [inline accounts/pkg/command/rebuild_index.go:15]: RebuildIndex?

File: accounts/pkg/indexer/index/disk/helper.go [hunk 2]
```diff
@@ -0,0 +1 @@
+package disk
```
Comment (butonic) [inline accounts/pkg/indexer/index/disk/helper.go:1]: unneeded? kill it with :fire:

File: accounts/pkg/storage/cs3.go [hunk 3]
```diff
@@ -20,18 +21,19 @@ import (
 	"github.com/cs3org/reva/pkg/token/manager/jwt"
 	"github.com/owncloud/ocis/accounts/pkg/config"
 	"github.com/owncloud/ocis/accounts/pkg/proto/v0"
+	olog "github.com/owncloud/ocis/ocis-pkg/log"
 	"google.golang.org/grpc/metadata"
 )
 
-// CS3Repo provides a cs3 implementation of the Repo interface
+// CS3Repo provides A cs3 implementation of the Repo interface
```
Comment (butonic) [inline accounts/pkg/storage/cs3.go:28]: lowercase is fine here

File: accounts/pkg/storage/cs3.go [hunk 4]
```diff
@@ -20,18 +21,19 @@ import (
...
 type CS3Repo struct {
 	cfg             *config.Config
 	tm              token.Manager
 	storageProvider provider.ProviderAPIClient
 	dataProvider    dataProviderClient // Used to create and download data via http, bypassing reva upload protocol
 }
 
-// NewCS3Repo creates a new cs3 repo
```
Comment (butonic) [inline accounts/pkg/storage/cs3.go:34]: lowercase is fine here

File: accounts/pkg/storage/cs3.go [hunk 5]
```diff
@@ -58,7 +60,7 @@ func NewCS3Repo(cfg *config.Config) (Repo, error) {
 	}, nil
 }
 
-// WriteAccount writes an account via cs3 and modifies the provided account (e.g. with a generated id).
+// WriteAccount writes an account via cs3 and modifies the provided account (e.G. with A generated id).
```
Comment (butonic) [inline accounts/pkg/storage/cs3.go:63]: ummm

File: accounts/pkg/storage/cs3.go [hunk 6]
```diff
@@ -138,7 +174,7 @@ func (r CS3Repo) DeleteAccount(ctx context.Context, id string) (err error) {
 	return nil
 }
 
-// WriteGroup writes a group via cs3 and modifies the provided group (e.g. with a generated id).
+// WriteGroup writes A group via cs3 and modifies the provided group (e.G. with A generated id).
```
Comment (butonic) [inline accounts/pkg/storage/cs3.go:177]: tumdidum

File: accounts/pkg/storage/cs3.go [hunk 7]
```diff
@@ -165,13 +201,47 @@ func (r CS3Repo) WriteGroup(ctx context.Context, g *proto.Group) (err error) {
 	return nil
 }
 
-// LoadGroup loads a group via cs3 by id and writes it to the provided group
+// LoadGroup loads A group via cs3 by id and writes it to the provided group
```
Comment (butonic) [inline accounts/pkg/storage/cs3.go:204]: did you replace `a` with `A`?

File: accounts/pkg/storage/cs3.go [hunk 8]
```diff
@@ -191,7 +261,7 @@ func (r CS3Repo) LoadGroup(ctx context.Context, id string, g *proto.Group) (err
 	return json.Unmarshal(b, &g)
 }
 
-// DeleteGroup deletes a group via cs3 by id
+// DeleteGroup deletes A group via cs3 by id
```
Comment (butonic) [inline accounts/pkg/storage/cs3.go:264]: aA

File: accounts/pkg/storage/cs3.go [hunk 9]
```diff
@@ -284,7 +354,7 @@ func MakeDirIfNotExist(ctx context.Context, sp provider.ProviderAPIClient, folde
 	return nil
 }
 
-// TODO: this is copied from proxy. Find a better solution or move it to ocis-pkg
+// TODO: this is copied from proxy. Find A better solution or move it to ocis-pkg
```
Comment (butonic) [inline accounts/pkg/storage/cs3.go:357]: aA

File: accounts/pkg/storage/disk.go [hunk 10]
```diff
@@ -15,13 +15,13 @@ import (
 
 var groupLock sync.Mutex
 
-// DiskRepo provides a local filesystem implementation of the Repo interface
+// DiskRepo provides A local filesystem implementation of the Repo interface
 type DiskRepo struct {
-	cfg       *config.Config
-	log       olog.Logger
```
Comment (butonic) [inline accounts/pkg/storage/disk.go:21]: and why the whitespace changes?

File: changelog/unreleased/accounts-index-rebuild.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
+Change: Rebuild index command for accounts
+
+Tags: accounts
+
+The index for the accounts service can now be rebuilt by running the cli command `./bin/ocis accounts rebuild`. It deletes all configured indices and rebuilds them from the documents found on storage.
```
Comment (butonic) [inline changelog/unreleased/accounts-index-rebuild.md:5]: and you added a ListAccounts & ListGroups call

**Review by butonic — CHANGES_REQUESTED** [review]

File: accounts/pkg/indexer/index/disk/helper.go (hunk 2 above)
Comment (butonic) [reply accounts/pkg/indexer/index/disk/helper.go:1]: ah ... here it comes

**Review by butonic — APPROVED** [review]
the only real naming issue is the RebuildIndex ... other than that I'm good

File: accounts/pkg/command/rebuild_index.go (hunk 1 above)
Comment (kulmann) [reply accounts/pkg/command/rebuild_index.go]: 👍

File: accounts/pkg/storage/cs3.go (hunk 7 above)
Comment (kulmann) [reply accounts/pkg/storage/cs3.go]: Refactoring in goland with too little intention did, yes. :-( same for `g` to `G`. Fixed them. Thanks for spotting!

**Comment by kulmann** [comment]
@refs we should take the time and write an integration test, at least with the disk implementation.

**Pull request merged**